applications/CompressiblePotentialFlowApplication/python_scripts/stochastic_potential_flow_response.py

```diff
@@ -0,0 +1,344 @@
+import KratosMultiphysics
+from KratosMultiphysics import Parameters, Logger
+import KratosMultiphysics.CompressiblePotentialFlowApplication as KCPFApp
+from KratosMultiphysics.response_functions.response_function_interface import ResponseFunctionInterface
+import KratosMultiphysics.CompressiblePotentialFlowApplication.potential_flow_analysis as potential_flow_analysis
+
+# Import Kratos, XMC, PyCOMPSs API
+import KratosMultiphysics.MultilevelMonteCarloApplication
+import xmc
+import xmc.methodDefs_momentEstimator.computeCentralMoments as mdccm
+from exaqute import get_value_from_remote
+import json
+
+def _GetModelPart(model, solver_settings):
+    #TODO can be removed once model is fully available
+    model_part_name = solver_settings["model_part_name"].GetString()
+    if not model.HasModelPart(model_part_name):
+        model_part = model.CreateModelPart(model_part_name, 2)
+        domain_size = solver_settings["domain_size"].GetInt()
+        if domain_size < 0:
+            raise Exception('Please specify a "domain_size" >= 0!')
+        model_part.ProcessInfo.SetValue(KratosMultiphysics.DOMAIN_SIZE, domain_size)
+    else:
+        model_part = model.GetModelPart(model_part_name)
+
+    return model_part
+
+class AdjointResponseFunction(ResponseFunctionInterface):
+
+    def __init__(self, identifier, response_settings, model):
+        self.identifier = identifier
+        self.response_settings = response_settings
+        self.xmc_settings_path = response_settings["xmc_settings"].GetString()
+        self.design_surface_sub_model_part_name = response_settings["design_surface_sub_model_part_name"].GetString()
+        self.auxiliary_mdpa_path = response_settings["auxiliary_mdpa_path"].GetString()
+        # Create the primal solver
+        with open(self.response_settings["primal_settings"].GetString(),'r') as parameter_file:
+            primal_parameters = Parameters( parameter_file.read() )
+
+        primal_parameters = _CheckParameters(primal_parameters)
+        if primal_parameters.Has("adjoint_parameters_path"):
+            primal_parameters["adjoint_parameters_path"].SetString(self.response_settings["adjoint_settings"].GetString())
+        else:
+            primal_parameters.AddString("adjoint_parameters_path", self.response_settings["adjoint_settings"].GetString())
+        if primal_parameters.Has("design_surface_sub_model_part_name"):
+            primal_parameters["design_surface_sub_model_part_name"].SetString(self.design_surface_sub_model_part_name)
+        else:
+            primal_parameters.AddString("design_surface_sub_model_part_name", self.design_surface_sub_model_part_name)
+        open(self.response_settings["primal_settings"].GetString(), 'w').write(primal_parameters.PrettyPrintJsonString())
+
+        self.primal_model_part = _GetModelPart(model, primal_parameters["solver_settings"])
+
+        self.primal_analysis = potential_flow_analysis.PotentialFlowAnalysis(model, primal_parameters)
```

Comment:
True, in fact it was possible to get rid of both! Since this class (`AdjointResponseFunction`) only gathers info, it can all be stored in a single model_part which is read from the input model, and the shape sensitivities can be stored in the non historical database (I dont need to go through the AdjointAnalysis either here).